GitUI/ScriptsEngine/FormFilePrompt.cs

```diff
@@ -64,4 +64,17 @@ private void Translate()
             Translator.Translate(this, AppSettings.CurrentTranslation);
         }
     }
+
+    public class FilePromptCreator : IFilePromptCreator
+    {
+        public IUserInputPrompt Create()
+        {
+            return new FormFilePrompt();
+        }
+    }
+
+    public interface IFilePromptCreator
+    {
+        IUserInputPrompt Create();
+    }
```

Comment:
I doubt that @RussKie will like this as it is.
❔
- Interface first
- Move to separate file `FilePromptCreator.cs`.
- Register IFilePromptCreator in Service Registry
- Usage: `uiCommands.GetRequiredService<IFilePromptCreator>().Create()`

Reply:
Indeed :)

Reply:
The concrete implementation should also remain non-public, as it should be reachable only via the interface registration.